### ISSUE TITLE
Examples of parsing exception handling are not actually handling exceptions

### ISSUE THREAD

**Issue by flytrex-vadim** [semuconsulting/pynmeagps#12]
![image](https://user-images.githubusercontent.com/20318135/149506636-4676eb48-d51e-4c43-8481-a78e9d55f7e6.png)
This is a pattern frequently seen in examples of box pyubx2 and pynmeagps. It seems to imply the intention of handling parsing errors in the except clause.
However, parsing exceptions are thrown from the next() iterator call which happens in line 74, the for loop, outside of the try-except block. So the except handler and the continue statement are never executed.

Here's my snippet with the corresponding call stack at the point of NMEATypeError exception throw:
![image](https://user-images.githubusercontent.com/20318135/149507070-72e7e70b-b9ca-46dc-a58e-70b8392e688b.png)

![image](https://user-images.githubusercontent.com/20318135/149507132-7af18257-1a45-492f-b665-c82bac9a2474.png)

One possible workaround is to use exception-hand wrapper around the iterator like this:

```
def exception_logging(gen):
    while True:
        try:
            yield next(gen)
        except StopIteration:
            break
        except Exception as e:
            print(e)

# usage: for (raw_data, parsed_data) in exception_logging(UBXReader(stream, ubxonly=False)):

```

**Comment by semudev2** [comment]
Ah OK I'm with you - you're referring to the iterator in `nmeafile.py`, for example. Yes, the exception handling should be improved in some of the iterator examples. On  it.

**Comment by flytrex-vadim** [comment]
I didn't run the code, but it looks like it indeed provides the intended functionality. 

Probably the wrapper could have been made part of the library to make it reusable to other users and reducing the amount of copy-paste boilerplate,

**Comment by semudev2** [comment]
I've added a provisional static `iterate()` wrapper to NMEAReader with an option to ignore or continue on error.

**Comment by flytrex-vadim** [comment]
How about adding a bit of configuration to the iterate ? For example, CLI application might not appreciate always printing the exception.

This is what I've tried in my code:

```
def exception_logging2(gen, error_handler = None):
    while True:
        try:
            yield next(gen)
        except StopIteration:
            break
        except Exception as e:
            if error_handler is not None:
                error_handler(e)

# usage example:
for (raw, msg) in exception_logging2(NMEAReader(stream, nmeaonly=False), lambda e: print(e)):

```

**Comment by semudev2** [comment]
good idea, done. will replicate in pyubx2 and explore other opportunities to do same elsewhere.
